### ISSUE TITLE
`preResolveTypes` lose `Maybe<>` typing

### ISSUE THREAD

**Issue by MarkKahn** [dotansimha/graphql-code-generator#2175]
Huge thanks for implementing `preResolveTypes`, it's beautiful!

It seems to be dropping all supplemental type info (maybe, array, ...anything else?) on primitive types though (strings, numbers, etc).  This data is being retained on the wrappers around nested object types:

```gql
export type someFragment = {
  key: string;        # should be `Maybe<string>`
  anotherKey: string; # should be string[]
  finalKey: Maybe<Array<{ key: string }>>;
}
```

Related to #2159

**Environment:**

- OS: OSX 10.14.5
- `@graphql-codegen/...`: `^0.14.0`
- NodeJS: 10.16.0

**Comment by dotansimha** [comment]
@MarkKahn I think I fixed it, can you please try? `1.4.1-alpha-1998a660.66`

**Comment by MarkKahn** [comment]
LGTM!  from 742 ts errors -> 0

**Comment by dotansimha** [comment]
Fixed in 1.5.0 🎉
